Feat/billing page

diegoanro22 opened the pull request:
## ✨ Descripción

Este Pull Request introduce la implementación completa de los **métodos de facturación (billing methods)** y sus correspondientes **modales de pago** dentro del módulo de Billing.  
Se centraliza la configuración de distintos métodos de pago, tanto locales como internacionales, y se integran mejoras visuales y de usabilidad en la interfaz del sistema.

* **Problema Resuelto / Característica Implementada:**
  * Implementación de modales específicos para distintos métodos de pago.
  * Estandarización de las configuraciones de facturación para proveedores.
  * Inclusión de descripciones multilenguaje y compatibilidad UI en Storybook.

* **Impacto:**
  * Mejora en la experiencia del usuario al gestionar opciones de pago.
  * Mayor escalabilidad del módulo de facturación, permitiendo añadir nuevos métodos fácilmente.
  * Incremento en la consistencia visual gracias a los componentes unificados en Storybook.

---

## 🚀 Cambios

* Implementación de **BankTransferModal** para transferencias bancarias.
* Creación de **CODModal** (Cash on Delivery) con verificación telefónica y pagos parciales.
* Adición de **CardPaymentModal** para pagos con tarjeta y selección de proveedor.
* Desarrollo de **CashPaymentModal** para pagos en efectivo con buenas prácticas UX.
* Incorporación de **InstallmentsModal** para planes de pago con aprobación automática.
* Creación de **PagaditoModal** para configuraciones del gateway Pagadito.
* Implementación de **PayPalModal** con soporte de client ID y secret.
* Integración de historias en **Storybook** para todos los modales de facturación.
* Eliminación de la selección de entorno (environment) en modales de pago.
* Actualización de descripciones con soporte multilenguaje.

---

## 🧪 Pruebas

* **Pruebas Manuales:**
  1. Navegar a la sección de *Billing Settings*.
  2. Abrir cada modal de pago y verificar su correcto renderizado y validación de campos.
  3. Probar la edición, guardado y cierre de configuraciones.
  4. Revisar visualmente las historias en **Storybook** (`npm run storybook`).
  5. Validar la responsividad en distintos dispositivos (iPhone 13, iPad Air).



---

## ✅ Checklist

Antes de enviar este Pull Request, por favor asegúrate de lo siguiente:

* [x] He leído la guía [CONTRIBUTING.md](https://github.com/GerardoFdez7/easystore-client/blob/main/CONTRIBUTING.md).
* [x] He realizado una auto-revisión de mi propio código.
* [x] He comentado mi código, particularmente en áreas difíciles de entender.
* [x] He realizado la documentación de los componentes en StoryBook si aplica.
* [x] Mis cambios no generan nuevas advertencias en la consola.
* [x] Todas las pruebas unitarias/de integración existentes y nuevas pasan localmente con mis cambios.
* [x] Los componentes nuevos o modificados siguen los principios de **Atomic Design**.
* [x] Se han considerado las optimizaciones de rendimiento de Next.js (lazy load, optimización de imágenes, etc.).

---

## 📸 Capturas de Pantalla (si aplica)

<img width="1222" height="532" alt="image" src="https://github.com/user-attachments/assets/2ef28411-7047-4107-b363-22e8bb72965d" />

<img width="1029" height="569" alt="image" src="https://github.com/user-attachments/assets/d2df5ea9-ca23-4a16-a748-cd75c291311b" />

<img width="1031" height="579" alt="image" src="https://github.com/user-attachments/assets/549b5c41-9dcc-4e12-9077-0280b4af8a1e" />


---

diegoanro22 committed:
feat(billing): add billing methods management and UI components

diegoanro22 committed:
feat(billing): add descriptions in multiple languages

diegoanro22 committed:
feat(billing): add modals for various payment methods

- Implement BankTransferModal for bank transfer configurations.
- Create CODModal for cash on delivery settings with options for partial payment and phone verification.
- Add CardPaymentModal for card payment configurations with provider and environment selection.
- Add CashPaymentModal for cash payment information and best practices.
- Introduce InstallmentsModal for managing installment plans with auto-approval feature.
- Built PagaditoModal for Pagadito payment configurations.
- Create PayPalModal for PayPal payment settings with client ID and secret.

diegoanro22 committed:
refactor(billing): remove environment selection from payment modals

diegoanro22 committed:
feat(billing): add storybook stories for various billing modals and templates

GerardoFdez7 commented on app/[locale]/components/shadcn/ui/textarea.tsx:
*NO MODIFICAR COMPONENTES PRIMITIVOS DE SHADCN* Crea otro componente en el que envuelvas este para implementar la funcionalidad que deseas.

GerardoFdez7 commented on stories/organisms/billing/MainBilling.stories.tsx:
```diff
@@ -0,0 +1,35 @@
+import MainBilling from '@organisms/billing/MainBilling';
+import { NextIntlClientProvider } from 'next-intl';
+
+const messages = {
```
No es necesario definir las traducciones en storybook, este ya tiene el mismo contexto que la aplicación.

GerardoFdez7 commented on stories/molecules/billing/BankTransferModal.stories.tsx:
```diff
@@ -0,0 +1,49 @@
+import BankTransferModal from '@molecules/billing/BankTransferModal';
+import { NextIntlClientProvider } from 'next-intl';
+
+const messages = {
```
No es necesario definir las traducciones en storybook, este ya tiene el mismo contexto que la aplicación.

GerardoFdez7 commented on stories/molecules/billing/CODModal.stories.tsx:
```diff
@@ -0,0 +1,47 @@
+import CODModal from '@molecules/billing/CODModal';
+import { NextIntlClientProvider } from 'next-intl';
+
+const messages = {
```
No es necesario definir las traducciones en storybook, este ya tiene el mismo contexto que la aplicación.

GerardoFdez7 commented on stories/molecules/billing/CashPaymentModal.stories.tsx:
```diff
@@ -0,0 +1,39 @@
+import CashPaymentModal from '@molecules/billing/CashPaymentModal';
+import { NextIntlClientProvider } from 'next-intl';
+
+const messages = {
```
No es necesario definir las traducciones en storybook, este ya tiene el mismo contexto que la aplicación.

GerardoFdez7 commented on app/[locale]/components/molecules/billing/BillingMethodCard.tsx:
```diff
@@ -0,0 +1,115 @@
+  cards: <CreditCard className="h-7 w-7" aria-hidden="true" />,
+  cash: <Banknote className="h-7 w-7" aria-hidden="true" />,
+  bank_transfer: <Landmark className="h-7 w-7" aria-hidden="true" />,
+  paypal: <Wallet className="h-7 w-7" aria-hidden="true" />,
```
Usa el icono de paypal aunque tengas que importarlo o definirlo de algún lugar que no sea lucide, solo asegurate de que funcione en ambos temas y tenga el mismo tamaño

GerardoFdez7 commented on app/[locale]/components/organisms/billing/MainBilling.tsx:
```diff
@@ -0,0 +1,148 @@
+
+  return (
+    <main className="flex w-full flex-col gap-6 px-2 sm:px-4 xl:mx-auto">
+      <section className="text-muted-foreground text-xl">
```
Usa la etiqueta h2 y el mismo estilo que el subtitulo en el dashboard: `className="text-text text-sm sm:text-lg"`

GerardoFdez7 commented on app/[locale]/components/organisms/billing/MainBilling.tsx:
```diff
@@ -0,0 +1,148 @@
+        })}
+      </section>
+
+      {/* Modales */}
```
Ingles

GerardoFdez7 commented on app/[locale]/components/molecules/billing/CODModal.tsx:
```diff
@@ -0,0 +1,167 @@
+                step="0.01"
+              />
+              <Button variant="outline" className="w-20">
+                USD
```
Usar la variable de entorno NEXT_PUBLIC_DEFAULT_CURRENCY

GerardoFdez7 commented on app/[locale]/components/organisms/billing/MainBilling.tsx:
```diff
@@ -0,0 +1,148 @@
+          }))
+        }
+      />
+      <CODModal
```
En la descripción de este metodo debemos describir que es pago contra entrega *con tarjeta*, ya que el efectivo cumple la misma función de pago contra entrega

GerardoFdez7 reviewed (changes requested):
Buen trabajo, me gustó como rehusaste BillingMethodCard para todos los metodos. Solo faltarían estas cositas:
- [ ]  Quitar todos los asteriscos en los Dialogs, ya que como sabes usaremos zod para validar que estén los campos requeridos
- [ ] Cambiar los placeholders de todos los campos para que muestren un ejemplo literal de lo que el usuario puede ingresar, sin usar eg, ej, etc.
- [ ] Agregar el icono Wallets en el dashboard debajo de Customers para que nos redirija a Billing

diegoanro22 committed:
feat(billing): add Pagadito configuration modal and translations

diegoanro22 committed:
feat(billing): :sparkles: refactor billing components and update translations

- Update BillingMethodCard to simplify state handling and adjusted styles for consistency.
- Modifie CODModal to use environment variable for currency display.
- Clean up CardPaymentModal by removing unnecessary labels and adjusting styles.
- Refine InstallmentsModal and PagaditoModal for better UI consistency.
- Enhance PayPalModal by removing unnecessary asterisk indicators.
- Update CarouselMedia to standardize max-width styling.
- Improve MainBilling layout by changing subtitle to a heading.
- Add billing link to Sidebar for better navigation.
- Add new translation keys for billing-related terms in English, Spanish, French, Italian, and Portuguese.
- Remove local message providers from Storybook stories for billing components to streamline testing.

diegoanro22 committed:
feat(billing): update Paypal icon
